timemachine/cpp/src/neighborlist.cu

```diff
@@ -76,34 +70,46 @@ void Neighborlist<RealType>::compute_block_bounds_host(
     double *d_coords = gpuErrchkCudaMallocAndCopy(h_coords, N * 3 * sizeof(double));
     double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3 * 3 * sizeof(double));
 
-    this->compute_block_bounds_device(N, D, d_coords, d_box, static_cast<cudaStream_t>(0));
+    this->compute_block_bounds_device(N, 0, D, d_coords, nullptr, d_box, static_cast<cudaStream_t>(0));
     gpuErrchk(cudaDeviceSynchronize());
 
     gpuErrchk(cudaMemcpy(
-        h_bb_ctrs, d_block_bounds_ctr_, this->B() * 3 * sizeof(*d_block_bounds_ctr_), cudaMemcpyDeviceToHost));
+        h_bb_ctrs, d_col_block_bounds_ctr_, this->B() * 3 * sizeof(*d_col_block_bounds_ctr_), cudaMemcpyDeviceToHost));
     gpuErrchk(cudaMemcpy(
-        h_bb_exts, d_block_bounds_ext_, this->B() * 3 * sizeof(*d_block_bounds_ext_), cudaMemcpyDeviceToHost));
+        h_bb_exts, d_col_block_bounds_ext_, this->B() * 3 * sizeof(*d_col_block_bounds_ext_), cudaMemcpyDeviceToHost));
 
     gpuErrchk(cudaFree(d_coords));
     gpuErrchk(cudaFree(d_box));
 }
 
 template <typename RealType>
-std::vector<std::vector<int>>
-Neighborlist<RealType>::get_nblist_host(int N, const double *h_coords, const double *h_box, const double cutoff) {
+std::vector<std::vector<int>> Neighborlist<RealType>::get_nblist_host(
+    int N, int K, const double *h_column_coords, const double *h_row_coords, const double *h_box, const double cutoff) {
 
-    // assert(N==N_);
+    assert(N + K == N_ + K_);
+    assert(K == K_);
+
+    double *d_col_coords = gpuErrchkCudaMallocAndCopy(h_column_coords, N * 3 * sizeof(double));
+
+    double *d_row_coords;
+    if (K > 0) {
+        d_row_coords = gpuErrchkCudaMallocAndCopy(h_row_coords, K * 3 * sizeof(double));
+    } else {
+        d_row_coords = d_col_coords;
+    }
 
-    double *d_coords = gpuErrchkCudaMallocAndCopy(h_coords, N * 3 * sizeof(double));
     double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3 * 3 * sizeof(double));
 
-    this->build_nblist_device(N, d_coords, d_box, cutoff, static_cast<cudaStream_t>(0));
+    this->build_nblist_device(N, K, d_col_coords, d_row_coords, d_box, cutoff, static_cast<cudaStream_t>(0));
 
     cudaDeviceSynchronize();
-    const int B = this->B(); //(N+32-1)/32;
+    const int tpb = 32;
```

Comment:
Not specific to this PR, but the definition `tpb = 32` seems replicated in a lot of places - would it make sense to declare as a global (or at least file-level) constant?

Reply:
We definitely should and its made worse that in some places tpb is used as a substitute for block size rather than warpsize